pmd_beamphysics/wavefront.py

```diff
@@ -0,0 +1,2084 @@
+            Add isophase contour to the phase plot.
+        figsize : (float, float), optional
+            Figure size for the axes.
+            Defaults to Matplotlib's `rcParams["figure.figsize"]``.
+        axs : List[matplotlib.axes.Axes], optional
+            Plot the data in the provided matplotlib Axes.
+            Creates a new figure and Axes if not specified.
+        cmap : str, default="viridis"
+            Color map to use.
+        nrows : int, default=2
+            Number of rows for the plot.
+        ncols : int, default=2
+            Number of columns for the plot.
+        save : pathlib.Path or str, optional
+            Save the plot to the given filename.
+        xlim : (float, float), optional
+            X axis limits.
+        ylim : (float, float), optional
+            Y axis limits.
+        tight_layout : bool, default=True
+            Set a tight layout.
+        transpose : bool, default=False
+            Transpose the data for plotting.
+
+        Returns
+        -------
+        Figure
+        list of Axes
+        """
+
+        if ax is None:
+            fig, ax = plt.subplots(1, 1, figsize=figsize)
+        else:
+            fig = ax.get_figure()
+
+        assert ax is not None
+
+        try:
+            r_or_k, xidx, yidx = projection_key_to_indices[projection]
+        except KeyError:
+            raise ValueError(
+                f"Unsupported projection: {projection} choose from {list(projection_key_to_indices)}"
+            ) from None
+
+        rspace = r_or_k == "rspace"
+        axis_indices = (xidx, yidx)
+
+        if rspace:
+            mesh_data = self.rmesh
+            labels = [_rspace_labels[idx] for idx in axis_indices]
+            domain = [self.rspace_domain[idx] for idx in axis_indices]
+            units = ["m", "m"]
+        else:
+            mesh_data = self.kmesh
+            labels = [_kspace_labels[idx] for idx in axis_indices]
+            domain = [self.kspace_domain[idx] for idx in axis_indices]
+            units = ["rad", "rad"]
+
+        domain_x, _scale, x_unit_prefix = nice_array(domain[0])
+        domain_y, _scale, y_unit_prefix = nice_array(domain[1])
+        extent = (domain_x[0], domain_x[-1], domain_y[-1], domain_y[0])
+
+        if transpose:
+            mesh_data = mesh_data.T
+            labels = tuple(reversed(labels))
+            extent = tuple(reversed(extent))
+            units = tuple(reversed(units))
+
+        sum_axis = tuple(
+            axis for axis in range(mesh_data.ndim) if axis not in axis_indices
+        )
+
+        assert len(sum_axis) == 1
+        sum_axis = sum_axis[0]
+
+        def plot(dat, title: str):
+            # TODO: balticfish will double-check
+            # _z_min, z_max = self.ranges[sum_axis]
+            # dz = self.grid_spacing[sum_axis]
+            # dat = np.sum(dat, axis=sum_axis) * dz / (2.0 * z_max)
+            img = ax.imshow(np.mean(dat, axis=sum_axis), cmap=cmap, extent=extent)
```

Comment:
`sum_axis` implies a sum, but here `np.mean` is called.

For power, it should probably be a sum. Others are questionable.